libs/libc/signal/sig_addset.c

```diff
@@ -79,3 +78,37 @@ int sigaddset(FAR sigset_t *set, int signo)
       return OK;
     }
 }
+
+/****************************************************************************
+ * Name: sigaddset
+ *
+ * Description:
+ *   This function adds the signal specified by signo to the signal set
+ *   specified by set.
+ *
+ * Input Parameters:
+ *   set - Signal set to add signal to
+ *   signo - Signal to add
+ *
+ * Returned Value:
+ *   0 (OK), or -1 (ERROR) if the signal number is invalid.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int sigaddset(FAR sigset_t *set, int signo)
+{
+  int ret;
+
+  /* Let nxsig_addset do all the work. */
+
+  ret = nxsig_addset(set, signo);
+  if (ret < 0)
+    {
+      set_errno(EINVAL);
```

Comment:
In the general case, it should set_errno(-ret);  In this case, -EINVAL should be the only value returned by nxsig_addset() so it does not matter.

Reply:
I kept EINVAL as it is the only valid errno value here.